src/quadtree.h

```diff
@@ -0,0 +1,65 @@
+#include "otpch.h"
+
+#include "creature.h"
+#include "spectators.h"
+
+namespace tfs::quadtree {
+void find(uint16_t start_x, uint16_t start_y, uint16_t end_x, uint16_t end_y, SpectatorVec& spectators,
+          std::function<bool(Creature*)> comparasion);
+Tile* find_tile(uint16_t x, uint16_t y, uint8_t z);
+void create_tile(uint16_t x, uint16_t y, uint8_t z, Tile* tile);
+void move_creature(uint16_t old_x, uint16_t old_y, uint8_t old_z, uint16_t x, uint16_t y, uint8_t z,
+                   Creature* creature);
+void insert_creature(uint16_t x, uint16_t y, uint8_t z, Creature* creature);
+void remove_creature(uint16_t x, uint16_t y, uint8_t z, Creature* creature);
```

Comment:
Can we use `Position` rather than 3 ints for xyz?

Reply:
Initially, I used pos, but to maintain compatibility with the `Map` class, I used x, y, and z. I would prefer if `Map` were also changed to support only Position. See https://github.com/otland/forgottenserver/blob/244465537b57230aa6824b390d979d2fe1a8eec2/src/map.h#L177-L190